pkg/kopia/cmd/commands.go

```diff
@@ -0,0 +1,486 @@
+func SnapshotExpireCommand(encryptionKey, rootID, configFilePath, logDirectory string, mustDelete bool) []string {
+	return stringSliceCommand(snapshotExpireCommand(encryptionKey, rootID, configFilePath, logDirectory, mustDelete))
+}
+
+func snapshotExpireCommand(encryptionKey, rootID, configFilePath, logDirectory string, mustDelete bool) logsafe.Cmd {
+	args := kopiaArgs(encryptionKey, configFilePath, logDirectory, false)
+	args = args.AppendLoggable(snapshotSubCommand, expireSubCommand, rootID)
+	if mustDelete {
+		args = args.AppendLoggable(deleteFlag)
+	}
+
+	return args
+}
+
+// SnapshotRestoreCommand returns kopia command restoring snapshots with given snap ID
+func SnapshotRestoreCommand(encryptionKey, snapID, targetPath, configFilePath, logDirectory string, sparseRestore bool) []string {
+	return stringSliceCommand(snapshotRestoreCommand(encryptionKey, snapID, targetPath, configFilePath, logDirectory, sparseRestore))
+}
+
+func snapshotRestoreCommand(encryptionKey, snapID, targetPath, configFilePath, logDirectory string, sparseRestore bool) logsafe.Cmd {
+	args := kopiaArgs(encryptionKey, configFilePath, logDirectory, false)
+	args = args.AppendLoggable(snapshotSubCommand, restoreSubCommand, snapID, targetPath)
+	if sparseRestore {
+		args = args.AppendLoggable(sparseFlag)
+	}
+
+	return args
+}
+
+// RestoreCommand returns the kopia command for restoring root of a snapshot with given root ID
+func RestoreCommand(encryptionKey, rootID, targetPath, configFilePath, logDirectory string) []string {
+	return stringSliceCommand(restoreCommand(encryptionKey, rootID, targetPath, configFilePath, logDirectory))
+}
+
+func restoreCommand(encryptionKey, rootID, targetPath, configFilePath, logDirectory string) logsafe.Cmd {
+	args := kopiaArgs(encryptionKey, configFilePath, logDirectory, false)
+	args = args.AppendLoggable(restoreSubCommand, rootID, targetPath)
+
+	return args
+}
+
+// DeleteCommand returns the kopia command for deleting a snapshot with given snapshot ID
+func DeleteCommand(encryptionKey, snapID, configFilePath, logDirectory string) []string {
+	return stringSliceCommand(deleteCommand(encryptionKey, snapID, configFilePath, logDirectory))
+}
+
+func deleteCommand(encryptionKey, snapID, configFilePath, logDirectory string) logsafe.Cmd {
+	args := kopiaArgs(encryptionKey, configFilePath, logDirectory, false)
+	args = args.AppendLoggable(snapshotSubCommand, deleteSubCommand, snapID, unsafeIgnoreSourceFlag)
+
+	return args
+}
+
+// SnapshotGCCommand returns the kopia command for issuing kopia snapshot gc
+func SnapshotGCCommand(encryptionKey, configFilePath, logDirectory string) []string {
+	return stringSliceCommand(snapshotGCCommand(encryptionKey, configFilePath, logDirectory))
+}
+
+func snapshotGCCommand(encryptionKey, configFilePath, logDirectory string) logsafe.Cmd {
+	args := kopiaArgs(encryptionKey, configFilePath, logDirectory, false)
+	args = args.AppendLoggable(snapshotSubCommand, gcSubCommand, deleteFlag)
+
+	return args
+}
```

Comment:
```suggestion
// SnapshotGCCommand returns the kopia command for issuing kopia snapshot gc
func SnapshotGCCommand(encryptionKey, configFilePath, logDirectory string) []string {
	args := kopiaArgs(encryptionKey, configFilePath, logDirectory, false)
	args = args.AppendLoggable(snapshotSubCommand, gcSubCommand, deleteFlag)
	return stringSliceCommand(args)
}
```

I don't see any issue in just combining both the functions since pattern seems same everywhere. Same for all the other wrappers.

Reply:
Resolved in foll commits 
- https://github.com/kanisterio/kanister/pull/1552/commits/878ac67682e949c1164c79d348eaec97360990c0 (in PR C)
- https://github.com/kanisterio/kanister/pull/1557/commits/8096870fc930501ba248119adcb636c4505966e7 (in PR D)